app/src/main/java/com/example/marcinwisniewski/intivetrainingapp/movielist/viewmodel/ListViewModel.kt

```diff
@@ -1,24 +1,30 @@
 package com.example.marcinwisniewski.intivetrainingapp.movielist.viewmodel
 
+import android.arch.lifecycle.LiveData
 import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.Transformations
 import android.arch.lifecycle.ViewModel
 import com.example.marcinwisniewski.intivetrainingapp.movielist.model.Movie
-import com.example.marcinwisniewski.intivetrainingapp.movielist.model.MovieProvider
+import com.example.marcinwisniewski.intivetrainingapp.movielist.model.MovieRepository
 import io.reactivex.android.schedulers.AndroidSchedulers
 import io.reactivex.schedulers.Schedulers
 import javax.inject.Inject
 
-class ListViewModel @Inject constructor(private val movieRepository: MovieProvider) : ViewModel() {
-    fun getMovies(): MutableLiveData<List<Movie>> {
-        // TODO: create handle error mechanism
-        val moviesList = MutableLiveData<List<Movie>>()
-        movieRepository.getMoviesResponse()
+class ListViewModel @Inject constructor(private val movieRepository: MovieRepository) : ViewModel() {
```

Comment:
Too generic name of ViewModel for Movies

Reply:
Changed to MovieListViewModel